src/sbvr-api/compile-auth.ts

```diff
@@ -0,0 +1,51 @@
+import type { Filter, PinejsClientCore, Resource } from 'pinejs-client-core';
+
+const asPermission = (compiledFilter: string): string => {
+	return compiledFilter
+		.split('$filter=')[1]
+		.replace(/Auth\.canAccess\(\)/g, 'canAccess()');
+};
+
+export const getAuthCompiler = <M extends { [key in keyof M]: Resource }>({
+	pinejsClient,
+	modelName,
+}: {
+	pinejsClient: PinejsClientCore<M>;
+	modelName: string;
+}) => {
+	if (pinejsClient == null) {
+		throw new Error('The pinejsClient must be specified.');
+	}
+
+	if (modelName == null) {
+		throw new Error('The modelName must be specified.');
+	}
+
+	return {
+		compileAuth: <TResource extends string & keyof M>({
+			resource,
+			access,
+			options,
+		}: {
+			resource: TResource;
+			access: string;
+			options?: {
+				$filter: Filter<M[TResource]['Read']>;
+			};
```

Comment:
It's probably clearer as
```suggestion
			options?: Pick<ODataOptions<M[TResource]['Read']>, '$filter'>;
```
to show that it's an ODataOptions object, but just the `$filter`